src/prefect/client/client.py

```diff
@@ -573,8 +573,8 @@ def register(
                 "Flows with required parameters can not be scheduled automatically."
             )
         if any(e.key for e in flow.edges) and flow.result_handler is None:
-            raise ClientError(
-                "Flows are required to have a result handler for storing inputs and outputs."
+            warnings.warn(
+                "It is strongly recommended you include a result handler on your Flow for certain Cloud features to work properly."
```

Comment:
@jlowin any thoughts on how to improve this message?

Reply:
It may be worth making the side effects more explicit, such as `No result handler was specified on your Flow. Cloud features such as input caching and task resuming may not work properly.`

Reply:
Yea that's a good call out